gluonnlp/model/char_encoder.py

```diff
@@ -0,0 +1,144 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Convolutional character encoder."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+__all__ = ['CharacterEncoder']
+
+from mxnet import gluon, nd
+from mxnet.gluon import nn
+
+from .highway import Highway
+
+
+class CharacterEncoder(gluon.Block):
+    r"""Convolutional character encoder.
+
+    We implement the convolutional character encoder proposed in the following work::
+
+        @inproceedings{kim2016character,
+         title={Character-Aware Neural Language Models.},
+         author={Kim, Yoon and Jernite, Yacine and Sontag, David and Rush, Alexander M},
+         booktitle={AAAI},
+         pages={2741--2749},
+         year={2016}
+        }
+
+    Parameters
+    ----------
+    embed_size : int
+        The input dimension to the encoder.
+    num_filters: int
+        The output dimension for each convolutional layer,
+        which is the number of the filters learned by the layer.
+    ngram_filter_sizes: Tuple[int]
+        The size of each convolutional layer,
+        and len(ngram_filter_sizes) equals to the number of convolutional layers.
+    conv_layer_activation: nn.Activation
+        Activation function to be used after convolutional layer.
+    num_highway: int
+        The number of layers of the Highway layer.
+    output_size: int
+        The output dimension after conducting the convolutions and max pooling,
+        and apply highways, as well as linear projection.
```

Comment:
grammar: ... after conducting convolutions and max pooling, and applying highways ...

Reply:
Thanks